osgar/logger.py

```diff
@@ -159,10 +173,16 @@ def __enter__(self):
     def __exit__(self, exc_type, exc_val, exc_tb):
         self.close()
 
+    def __next__(self):
+        return next(self.gen)
+
+    def __iter__(self):
+        return self
```

Comment:
One small nit - could we return `self.gen` here instead of `self`? That should allow us to remove the `__next__` method above while nothing would change on the usage side.

Reply:
Scratch that. It would not allow us to call `next` directly on the opened log without first calling `iter` which we do frequently. Sorry.